rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/cache/RedisHashCache.java

```diff
@@ -475,52 +422,38 @@ public void remove(final KeyType key) {
             log.warn("{}: Remove requested for null key", cacheName);
             return;
         }
-        rdbi.withHandle(new Callback<Void>() {
-            @Override
-            public Void run(Handle handle) {
+        rdbi.consumeHandle(handle -> {
                 Pipeline pipeline = handle.jedis().pipelined();
                 // remove the cached value (if any)
                 pipeline.hdel(cacheKey, itemKey(key));
                 // clear out missing cache key (if any)
                 pipeline.srem(cacheMissingKey(), keyTypeSerializationHelper.encode(key));
                 pipeline.sync();
-                return null;
-            }
         });
     }
 
-    private Map<KeyType, ValueType> uniqueIndex(Iterable<ValueType> values) {
-        return FluentIterable.from(values)
-                             .filter(new Predicate<ValueType>() {
-                                 @Override
-                                 public boolean apply(@Nullable ValueType value) {
-                                     if (value == null) {
-                                         log.warn("{}: Omitting a null value from the result set.", cacheName);
-                                     }
-                                     return value != null;
-                                 }
-                             })
-                             .uniqueIndex(new Function<ValueType, KeyType>() {
-                                 @Override
-                                 public KeyType apply(ValueType value) {
-                                     return keyFromValue(value);
-                                 }
-                             });
+    private Map<KeyType, ValueType> uniqueIndex(Collection<ValueType> values) {
+        return values.stream()
+                     .filter(value -> {
+                         if (value == null) {
+                             log.warn("{}: Omitting a null value from the result set.", cacheName);
+                         }
+                         return value != null;
+                     })
+                     .collect(Collectors.toMap(this::keyFromValue,
+                                               Function.identity(),
+                                               (a, b) -> a));
```

Comment:
i didn't know about the possibility of illegalstateexception here. incredibly useful to know, thanks for doing it.

Reply:
It's burned me once or twice when I didn't think about it. I almost never want the default behavior of throwing on duplicate.